sympy/series/limits.py

```diff
@@ -146,15 +146,14 @@ def doit(self, **hints):
 
         if e.is_Mul:
             if abs(z0) is S.Infinity:
-                # XXX todo: this should probably be stated in the
-                # negative -- i.e. to exclude expressions that should
-                # not be handled this way but I'm not sure what that
-                # condition is; when ok is True it means that the leading
-                # term approach is going to succeed (hopefully)
+                ztemp = Symbol("z",real = True)
```

Comment:
@jksuom cases like  `(limit((x**1000/((x + 1)**1000 + exp(-x))), x, oo))` in `test_issue_5955()` the gruntz algorithm was running in an almost tending loop, so I tried adjusting the conditions for `ok`. `ok` which was specific to polyomials would have returned `False` but with this will return '`True'.